golem/managers/proposal/plugins/buffer.py

```diff
@@ -99,32 +100,31 @@ async def _request_items(self) -> None:
 
     @trace_span(show_results=True)
     async def get_proposal(self) -> Proposal:
-        if not self._get_items_count():
-            logger.debug("No items to get, requesting fill")
-            await self._request_items()
+        if not self._get_buffered_proposals_count():
+            logger.debug("No proposals to get, requesting fill")
+            await self._request_proposals()
 
-        proposal = await self._get_item()
+        proposal = await self._get_buffered_proposal()
 
-        items_count = self._get_items_count()
-        if items_count < self._min_size:
+        proposals_count = self._get_buffered_proposals_count()
+        if proposals_count < self._min_size:
             logger.debug(
-                "Items count is now `%s` which is below min size `%d`, requesting fill",
-                items_count,
+                "Proposals count %d is below minimum size %d, requesting fill",
+                proposals_count,
                 self._min_size,
             )
-            await self._request_items()
+            await self._request_proposals()
         else:
             logger.debug(
-                "Target items is now `%s` which is not below min size `%d`, requesting fill not"
-                " needed",
-                items_count,
+                "Proposals count %d is not below minimum size %d, skipping fill",
+                proposals_count,
                 self._min_size,
             )
 
         return proposal
 
-    async def _get_item(self) -> Proposal:
+    async def _get_buffered_proposal(self) -> Proposal:
         return await self._buffer.get()
 
-    def _get_items_count(self) -> int:
+    def _get_buffered_proposals_count(self) -> int:
```

Comment:
same - is there a way to get a non-buffered proposals count?

```suggestion
    def _get_proposals_count(self) -> int:
```